simulation/library/libraries.py

```diff
@@ -102,19 +106,19 @@ def GetGoDirectory(self):
                     return f"{binaries_directory}/{binary_container}"
                 except:
                     pass
-        except:
+        except LibrariesNotFound:
             if self.raiseExceptionOnFail:
-                raise Exception("GoLang binaries not found for your operating system. "
-                                "Please either compile them for your operating system or disable GoLang usage "
-                                "in Simulation instantiation.")
+                raise LibrariesNotFound("GoLang libraries not found for your operating system. "
```

Comment:
Is this exception caught somewhere so that the python functions can be used instead? I think the default functionality should be that if Go is requested but the binary is not found, a warning is raised and python should be used

Reply:
Currently the instantiation of the Libraries object is passing `raiseExceptionOnFail=False`, so the exception will never be raised currently. I'm think that a try/catch block and getting rid of the `raiseExceptionOnFail ` flag would be a cleaner solution.